mbedtls/mbedtls.py

```diff
@@ -7,32 +7,17 @@
 
 def harness(dut: Dut, ctx: TestContext, result: TestResult):
     subresult = None
-    msg = []
     test_status = Status.OK
 
-    RESULT = r"(?P<name>.{67}\s)(?P<status>(PASS|----|FAILED))\r+\n"
-    MSG_LINE = r"  (?P<line>[^\r\n]+?)\r+\n"
+    RESULT = r"(?P<name>.{67}\s)(?P<status>(PASS|----))\r+\n"
+    RESULT_FAIL = r"(?P<name>.{67}\s)(?P<status>(FAILED))\r+\n (?P<line>[^\r\n]+?)\r+\n"
```

Comment:
IMHO multiline regexps are not a good idea, they might break in unexpected ways depending on how they are applied or might result in exponential execution time (eg. greedy approach to matching regexp might parse next test as a message line). If we can use single-line regexps + state machine (like previously) it's much better (and easier to understand at a glance).

Does my previously proposed solution not work in some cases?

Reply:
I agree, a single-line regex is better. I’ve just sent another approach.  The general issue is that by the time we collect the test completion time, we must already have a message, especially if the test failed. That's why add_subresult_obj was executed after the completion of the next test. This resulted in the test time being collected from the subsequent test.

Reply:
ok, I see what's the issue with my approach, the subresult in `trunner` is being added by-value instead of by-reference, probably the best solution would be to change that in test runner as it would simplify the harness implementation greatly.

Should be easy to fix, I can think for a clean solution for that soon.

Reply:
Please check if this solution would work  in regards to the messages printed from the test cases. - I've changed the API and applied a quickfix only for mbedtls: 
https://github.com/phoenix-rtos/phoenix-rtos-tests/tree/nalajcie/trunner

Reply:
So, I've tested it. Now, in the case of mbedtls, both the messages and the timing are correctly appended. The only issue is that if there's a fail and any fail messages follow it, the time between the end of the test case and the collection of the message will be added to the next test. But from what I see, it's only a maximum of 0.-0.15 seconds, so I would consider that an acceptable range

Reply:
Tanks for testing. I consider this a negligible time difference and IMHO the simplicity of the solution is worth it. 

I'll make my branch a PR, please add your commit to this branch (`nalajcie/trunner`) with necessary changes in busybox (yep, we would be PR co-authors ;)).